.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,63 @@
+name: Benchmark on Raspberry Pi 5
+
+on:
+  schedule: # Runs at midnight (0:00 UTC) every day
+    - cron: "0 0 * * *"
+  push:
+    branches: [main]
+  pull_request:
+
+env:
+  DIVAN_MAX_TIME: 30
+  CODSPEED_RUNNER_MODE: walltime
+
+jobs:
+  build:
+    name: Build benchmark binaries
+    runs-on: ubuntu-22.04-arm
```

Comment:
It's better to be explicit about what the compilation target is. If you compile this machine, which processors will the resulting executable run on? Which fancy instructions does it use? It may crash on a different processor.
Minor to this is optimization: microarchitecture (pipelines etc.) differ between processors and this can be quite a substantial difference in performance. We already see this in a ~20% performance difference between two skyscraper algorithms, with the fastest one being different on an M3 than on the RPi5!

Reply:
And it's a bit more involved to build the benchmarks without it (wrangling the benchmark binaries + handling the walltime/instrumentation modes)